src/webapi/Mapping/MappingHelper.cs

```diff
@@ -38,11 +47,20 @@ public static void PopulateEvaluationPK(object srcEvaluationObject, object dstEv
             };
             var srcObjType = srcEvaluationObject.GetType();
             var dstObjType = dstEvaluationObject.GetType();
+
+            var dstProperties = dstObjType.GetProperties().ToList();
+            var srcProperties = srcObjType.GetProperties().ToList();
             foreach (var pkCol in evalPKCols)
             {
-                dstObjType.GetProperty(pkCol).SetValue(
-                    dstEvaluationObject,
-                    srcObjType.GetProperty(pkCol).GetValue(srcEvaluationObject));
+                if (dstProperties.Exists(x => x.Name == pkCol) && srcProperties.Exists(x => x.Name == pkCol))
+                {
+                    // Disabled: the filters above solve this.
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
+                    dstObjType.GetProperty(pkCol).SetValue(
+                        dstEvaluationObject,
+                        srcObjType.GetProperty(pkCol).GetValue(srcEvaluationObject));
+#pragma warning restore CS8602 // Dereference of a possibly null reference.
+                }
```

Comment:
I thought about throwing an exception in the else case, and still might do so. "Shouldn't happen" though.